// @ai-sdk/vercel/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  createVercel: () => createVercel,
  vercel: () => vercel
});
module.exports = __toCommonJS(src_exports);

// src/vercel-provider.ts
var import_provider = require("@ai-sdk/provider");
var import_openai_compatible = require("@ai-sdk/openai-compatible");
var import_provider_utils = require("@ai-sdk/provider-utils");
function createVercel(options = {}) {
  var _a;
  const baseURL = (0, import_provider_utils.withoutTrailingSlash)(
    (_a = options.baseURL) != null ? _a : "https://api.v0.dev/v1"
  );
  const getHeaders = () => ({
    Authorization: `Bearer ${(0, import_provider_utils.loadApiKey)({
      apiKey: options.apiKey,
      environmentVariableName: "VERCEL_API_KEY",
      description: "Vercel"
    })}`,
    ...options.headers
  });
  const getCommonModelConfig = (modelType) => ({
    provider: `vercel.${modelType}`,
    url: ({ path }) => `${baseURL}${path}`,
    headers: getHeaders,
    fetch: options.fetch
  });
  const createChatModel = (modelId, settings = {}) => {
    return new import_openai_compatible.OpenAICompatibleChatLanguageModel(modelId, settings, {
      ...getCommonModelConfig("chat"),
      defaultObjectGenerationMode: "json"
    });
  };
  const provider = (modelId, settings) => createChatModel(modelId, settings);
  provider.chatModel = createChatModel;
  provider.languageModel = createChatModel;
  provider.textEmbeddingModel = (modelId) => {
    throw new import_provider.NoSuchModelError({ modelId, modelType: "textEmbeddingModel" });
  };
  return provider;
}
var vercel = createVercel();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createVercel,
  vercel
});
